jwst/pipeline/calwebb_coron3.py

```diff
@@ -46,17 +46,25 @@ class Coron3Pipeline(Pipeline):
 
     prefetch_references = False
 
+    # Main processing
     def process(self, user_input):
-        """Primary method for performing pipeline."""
+        """Primary method for performing pipeline.
+
+        Parameters
+        ----------
+        user_input: str, Level3 Association, or ~jwst.datamodels.DataModel
+            The exposure or association of exposures to process
+        """
         self.log.info('Starting calwebb_coron3 ...')
+        asn_exptypes = ['science', 'psf']
 
-        # Load the input association table
-        asn = self.load_as_level3_asn(user_input)
-        acid = asn.get('asn_id', '')
+        # Create a DM object using the association table
+        input_models = datamodels.open(user_input, asn_exptypes=asn_exptypes)
+        acid = input_models.meta.asn_table.asn_id
 
         # We assume there's one final product defined by the association
-        prod = asn['products'][0]
-        self.output_file = prod.get('name', self.output_file)
+        prod = input_models.meta.asn_table.products[0].instance
```

Comment:
Is the `.instance` needed here? It may be but I am surprised, since at this point `ans_table` should simply be pointing to the astropy table.

Reply:
That's the way it's also done in calwebb_spec3:
https://github.com/spacetelescope/jwst/blame/master/jwst/pipeline/calwebb_spec3.py#L117
but still not sure that proves it's really necessary. Blame suggests that we ask @jdavies-st